engine/samples/events/page.md

```diff
@@ -0,0 +1,28 @@
+# Events {#examples-engine-events}
+
+@brief Using the @ref cubos::core::ecs::EventReader and @ref cubos::core::ecs::EventWriter
+
+This example shows how the @ref cubos::core::ecs::EventReader and the @ref cubos::core::ecs::EventWriter can be used and configured, in a simple scene, to emit and read events. These are in the core, meaning, you don't need to add them to cubos.
+
+Firstly, we need to create and register the event we want to emit. Here, our event has one variable, however, you can give it any number of variables of any type you want.
+
+@snippet events/main.cpp Event struct
+@snippet events/main.cpp Adding event
+
+To receive these events, we can make a simple system, using the @ref cubos::core::ecs::EventReader and iterating through all the events it has. This will be the layout of all our reader systems (A, C, D).
```

Comment:
```suggestion
To receive these events, we can make a simple system which takes the @ref cubos::core::ecs::EventReader "EventReader" system argument and iterates through all the events it has. This will be the layout of all our reader systems (A, C, D).
```